src/backend/joanie/core/utils/webhooks.py

```diff
@@ -40,28 +40,34 @@ def synchronize_course_runs(serialized_course_runs):
             digestmod=hashlib.sha256,
         ).hexdigest()
 
-        response = session.post(
-            webhook["url"],
-            json=serialized_course_runs,
-            headers={"Authorization": f"SIG-HMAC-SHA256 {signature:s}"},
-            verify=bool(webhook.get("verify", True)),
-            timeout=3,
-        )
-
-        extra = {
-            "sent": json_course_runs,
-            "response": response.content,
-        }
-        # pylint: disable=no-member
-        if response.status_code == requests.codes.ok:
-            logger.info(
-                "Synchronization succeeded with %s",
+        try:
+            response = session.post(
                 webhook["url"],
-                extra=extra,
+                json=serialized_course_runs,
+                headers={"Authorization": f"SIG-HMAC-SHA256 {signature:s}"},
+                verify=bool(webhook.get("verify", True)),
+                timeout=3,
             )
-        else:
+
+            extra = {
+                "sent": json_course_runs,
+                "response": response.content,
+            }
+            # pylint: disable=no-member
+            if response.status_code == requests.codes.ok:
+                logger.info(
+                    "Synchronization succeeded with %s",
+                    webhook["url"],
+                    extra=extra,
+                )
+            else:
+                logger.error(
+                    "Synchronization failed with %s",
+                    webhook["url"],
+                    extra=extra,
+                )
+        except requests.exceptions.RetryError:
             logger.error(
-                "Synchronization failed with %s",
+                "Synchronization failed due to max retries exceeded with url %s",
                 webhook["url"],
-                extra=extra,
             )
```

Comment:
What do you think to put except above else statement to improve readability ?

Reply:
the code is attached with the try, doing so throws a local variable 'response' referenced before the assignment